atc/api/volumeserver/report.go

```diff
@@ -42,6 +43,23 @@ func (s *Server) ReportWorkerVolumes(w http.ResponseWriter, r *http.Request) {
 		"handles-count": len(handles),
 	})
 
+	numUnknownVolumes, err := s.repository.DestroyUnknownVolumes(workerName, handles)
+	if err != nil {
+		logger.Error("failed-to-destroy-unknown-volumes", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if numUnknownVolumes > 0 {
```

Comment:
(see comment on containers regarding having the metric emission outside the condition 😁 )